flamingpy/codes/surface_code.py

```diff
@@ -278,8 +281,14 @@ def __init__(
         polarity=None,
         backend="retworkx",
     ):
+        assert np.size(distance) in (1, 3), "distance must be a scalar or a 3-tuple"
```

Comment:
@ilan-tz should I throw a TypeError or ValueError here? I don't know what the convention is.

Reply:
It sounds like the spirit of this test should be to check if the correct type of object is input, which means you should raise a TypeError. If you then check whether the scalar is an int and the elements of the tuple are ints, you could also raise a ValueError if you wish.

Note: are we looking for any size-3 list-type thing or specifically a tuple? This will change how you do this check.

Reply:
Any list-type works, I'll do the checks accordingly 👍🏼

Reply:
Actually, it might be neater to ask for a tuple @ilan-tz. Let me know how you like the implementation in [this commit ](https://github.com/XanaduAI/flamingpy/pull/61/commits/a7c9fa3a315d55b225e845067bb7fe39c9f03bb4)

For reference: if I use typing directly with the input I can check the length too, but only if it's a tuple: https://stackoverflow.com/a/44833864/7658078